### ISSUE TITLE
An add_package("glm") in a target won't be exported to other target which contains a dependency on the target

### ISSUE THREAD

**Issue by HeBomou** [xmake-io/xmake#1109]
### Describe the bug

I have a static target. GLM is added to it as a package, and its headers are exposed by calling add_includedirs with {public = true}. Now if I add this target as a dependency to other target, glm won't be exposed.

Here goes an example to reproduce it

xmake.lua

``` lua
set_project(Test)
set_languages("c++17")

add_requires("glm 0.9.9+8", {alias = "glm"})

target("lib")
    set_kind("static")
    add_includedirs("lib/", {public = true})
    add_files("lib/*.cpp")
    add_packages("glm")

target("app")
    set_kind("binary")
    add_files("app/app.cpp")
    add_deps("lib")
```

lib/lib.h

``` cpp
#include <glm/vec3.hpp>
glm::vec3 mul(glm::vec3 vector, float factor);
```

lib/lib.cpp

``` cpp
#include <lib.h>
glm::vec3 mul(glm::vec3 vector, float factor) {
    return vector * factor;
}
```

app/app.cpp

``` cpp
#include <lib.h>

int main() {
    return 0;
}
```

### Expected behavior

The example should be compiled
GLM headers should be exposed to app.cpp

### Error output

``` zsh
% xmake -vD
[ 37%]: compiling.release lib/lib.cpp
"/usr/bin/xcrun -sdk macosx clang" -c -Qunused-arguments -arch x86_64 -mmacosx-version-min=11.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk -std=c++17 -Ilib -I/Users/hebo/.xmake/packages/g/glm/0.9.9+8/d639b7d6e3244216b403b39df5101abf/include -o build/.objs/lib/macosx/x86_64/release/lib/lib.cpp.o lib/lib.cpp
[ 50%]: compiling.release app/app.cpp
"/usr/bin/xcrun -sdk macosx clang" -c -Qunused-arguments -arch x86_64 -mmacosx-version-min=11.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk -std=c++17 -Ilib -o build/.objs/app/macosx/x86_64/release/app/app.cpp.o app/app.cpp
error: @programdir/modules/private/async/runjobs.lua:210: @programdir/modules/private/action/build/object.lua:75: @programdir/modules/core/tools/gcc.lua:473: In file included from app/app.cpp:1:
lib/lib.h:3:10: fatal error: 'glm/vec3.hpp' file not found
#include <glm/vec3.hpp>
         ^~~~~~~~~~~~~~
1 error generated.
stack traceback:
    [C]: in function 'error'
    [@programdir/core/base/os.lua:792]: in function 'raise'
    [@programdir/modules/core/tools/gcc.lua:473]: in function 'catch'
    [@programdir/core/sandbox/modules/try.lua:127]: in function 'try'
    [@programdir/modules/core/tools/gcc.lua:422]:
    [C]: in function 'compile'
    [@programdir/modules/private/action/build/object.lua:75]: in function 'script'
    [@programdir/modules/private/action/build/object.lua:88]: in function '_build_object'
    [@programdir/modules/private/action/build/object.lua:113]: in function 'jobfunc'
    [@programdir/modules/private/async/runjobs.lua:188]:
    [C]: in function 'trycall'
    [@programdir/core/sandbox/modules/try.lua:121]: in function 'try'
    [@programdir/modules/private/async/runjobs.lua:182]: in function 'cotask'
    [@programdir/core/base/scheduler.lua:317]:

stack traceback:
        [C]: in function 'error'
        @programdir/core/base/os.lua:792: in function 'raise'
        @programdir/modules/private/async/runjobs.lua:210: in function 'catch'
        @programdir/core/sandbox/modules/try.lua:127: in function 'try'
        @programdir/modules/private/async/runjobs.lua:182: in function 'cotask'
        @programdir/core/base/scheduler.lua:317: in function <@programdir/core/base/scheduler.lua:315>
```

### Related Environment

- xmake version: 2.3.8(brew)
- os: macOS 11.0.1
- target platform: macosx x86_64

### Additional context

Note that GLM is a header only library. And I found while calling add_requires XMake merely put its headers into the inlude directory

**Comment by xq114** [comment]
A simple (but not so straight forward) workaround is to add `add_packages("glm")` to target `app` as well. According to  https://github.com/xmake-io/xmake/issues/1104 a new feature may be added in future?

**Comment by HeBomou** [comment]
> A simple (but not so straight forward) workaround is to add `add_packages("glm")` to target `app` as well.

Well, it actually works

> According to #1104 a new feature may be added in future?

Maybe it's a bug? But it may also be automatically fixed if adding the feature metioned in #1104

**Comment by waruqi** [comment]
> Maybe it's a bug? But it may also be automatically fixed if adding the feature metioned in #1104

It is not a bug. By default includedirs is private, users need to manually configure {public = true} to export it. Of course add_packages does not support it yet.

I will support it in future versions, but currently you can only add glm to the app target.

**Comment by waruqi** [comment]
I have supported it. https://github.com/xmake-io/xmake/issues/1254
